### PR TITLE
Implemented search triggering when the search icon is clicked

### LINKED ISSUE #1292: Search Functionality Not Triggered by Search Icon
**Describe the bug**
The search function does not initiate when the search icon is clicked. It only activates when the Enter key is pressed. The desired behavior is for the search to begin when either the icon is clicked or the Enter key is used.

**To Reproduce**

1. Navigate to Angel Organization
2. Go to Posts Screen
3. Enter a search query in the search bar.
4. Click the search icon.
5. Observe that the search results do not change.

**Expected behavior**
Searching should even function when search icon is clicked not only just by enter key.

**Actual behavior**
Searching only functions when the Enter key is pressed.

**Video**

https://github.com/PalisadoesFoundation/talawa-admin/assets/112749383/4b64df0a-2819-4be4-9915-df71029bf697

**Additional details**

 This issue is not confined to the Posts screen; it pertains to all search bars within the application.

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by akhilender-bongirwar**
**What kind of change does this PR introduce?**
This PR introduces search functionality even by search icon click.

**Issue Number:**

Fixes #1292

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa-admin/assets/112749383/a4de808a-022d-47bc-aaac-9985e000761d

**If relevant, did you update the documentation?**
No

**Summary**
- Added code to implement search functionality even when the search icon is clicked.
- Enhanced search functionality by allowing search initiation through both pressing 'Enter' and clicking the search icon in the user portal, ensuring uniformity.
- Expanded test coverage to include all modified lines.
- Users can now utilize dual search methods by both pressing 'Enter' and clicking the search icon to obtain search results.

**Does this PR introduce a breaking change?**
No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
Please fix the failing test

**Comment by SiddheshKukade** [comment]
Please fix the failing tests @akhilender-bongirwar

**Comment by akhilender-bongirwar** [comment]
> Please fix the failing tests @akhilender-bongirwar

@SiddheshKukade, please review I have fixed all the failing tests and addressed all the missing lines in the code coverage report

File: src/screens/Requests/Requests.tsx [hunk 1]
```diff
@@ -239,22 +239,34 @@ const Requests = (): JSX.Element => {
   };
 
   /* istanbul ignore next */
-  const handleSearchByName = async (e: any): Promise<void> => {
+  const handleSearch = async (value: string): Promise<void> => {
```
Comment (aashimawadhwa) [inline src/screens/Requests/Requests.tsx:242]: please remove /* istanbul ignore next */ comment and resubmit the PR.

File: src/screens/UserPortal/People/People.tsx [hunk 2]
```diff
@@ -70,17 +69,24 @@ export default function people(): JSX.Element {
...
 
-    refetch(filter);
+  const handleSearchByBtnClick = (): void => {
+    const inputValue =
+      (document.getElementById('searchPeople') as HTMLInputElement)?.value ||
+      '';
+    handleSearch(inputValue);
   };
 
   /* istanbul ignore next */
```
Comment (aashimawadhwa) [inline src/screens/UserPortal/People/People.tsx:92]: remove this comment again

**Review by aashimawadhwa — DISMISSED** [review]

File: src/screens/Requests/Requests.tsx (hunk 1 above)
Comment (akhilender-bongirwar) [reply src/screens/Requests/Requests.tsx:242]: Sure, I will remove them.

**Comment by akhilender-bongirwar** [comment]
@SiddheshKukade @aashimawadhwa @beingnoble03 @yasharth291 please review

**Comment by akhilender-bongirwar** [comment]
@palisadoes, please review this PR.

**Review by SiddheshKukade — APPROVED** [review]

**Review by palisadoes — APPROVED** [review]

**Pull request merged**